docs/examples/codespaces.md

```diff
@@ -0,0 +1,111 @@
+
+Earthly has been tested with Codespaces on Github.com and in Visual Stuidio Code.
+
+### Resources
+
+ * [Codespaces Deep Dive](https://docs.github.com/en/codespaces/getting-started/deep-dive)
+ * [Codespaces Devcontainers](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers)
+ * [Install Earthly](https://earthly.dev/get-earthly)
+
+
+## Setup with Satellites (Recommended)
+
+With Satellites you can run Earthly in your devcontainer without needing to install docker or podman! This can allow you to have a more simplified setup as well as faster Earthly build times.
+
+If you don't already have an Earthly account or satellites read more about them here:
+- https://docs.earthly.dev/earthly-cloud/satellites
+
+Login here: 
+- https://cloud.earthly.dev/login
+
+
+How To:
+1. Update your devcontainer configuration in .devcontainer/devcontainer.json to use a Dockerfile
+
+    ```
+    {
+        "name": "My Devcontainer",
+
+        "build": {
+            "dockerfile": "Dockerfile"
+        },
+        ...
+        // YOUR OTHER CONFIG HERE
+
+        // You can also add on the Earthly syntax highlighting extension!
+        "customizations": {
+            "vscode": {
+                "extensions": [
+                    "earthly.earthfile-syntax-highlighting"
+                ]
+            }
+        },
+        ...
+    }
+    ```
+
+1. Update your Dockerfile to include installing Earthly and any other setup you may need
+    
+    `sudo /bin/sh -c 'wget https://github.com/earthly/earthly/releases/latest/download/earthly-linux-amd64 -O /usr/local/bin/earthly && chmod +x /usr/local/bin/earthly`
+    
+    _Note_: You don't need to install docker or podman, nor bootstrap earthly, when using satellites! Allowing you to further simplify your codespaces configuration
+1. Rebuild your container to include the installation
+1. Login `earthly account login --token {YOUR_TOKEN_HERE}`
+    
+    For how to handle secrets view Codespaces secret guide here: https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces
+1. Select your org `earthly org select {YOUR_ORG_NAME_HERE}`
+1. Launch your satellite: `earthly sat launch {YOUR_SAT_NAME_HERE}`
+1. Your Earthly commands will now run on your satellite!
+    
+    _Note_: It is recommended you login to your docker registry account to prevent rate limiting issues.
+
+
+## Setup with Docker (Using Devcontainer Feature)
+
+The Feature used in this example includes installing docker so there is no need to install it separately
+
+This example was run using the `mcr.microsoft.com/devcontainers/universal:2` image.
+
+1. Update your devcontainer configuration in .devcontainer/devcontainer.json in the root of your project 
+    ````
+    // YOUR OTHER CONFIG HERE
+    ...
+    ...
+    "features": {
+        "ghcr.io/shepherdjerred/devcontainers-features/earthly:1": {
```

Comment:
What does this part do? It seems to be relying on a third party container?

Reply:
It's a feature which is a set of installation code for Codespaces. In this case, yes it is a third party that created the feature for Earthly installation. That's a good point to bring up since we don't have control of what goes in to that feature so I'll update the Example to use the official features + how to install Earthly on top of that instead

Reply:
I'm reading around about this "feature" concept for the first time :D

Perhaps the best user experience would be if we support a feature like this one ourselves, and recommend it officially.

I'll check with Jerred Shepherd on Slack whether he'd be open to share his work as open-source and we can credit him.